airflow/contrib/hooks/opsgenie_alert_hook.py

```diff
@@ -0,0 +1,181 @@
+    :param description: Description field of the alert
+    :type description: str
+    :param responders: Teams, users, escalations and schedules that
+                      the alert will be routed to send notifications.
+    :type responders: list[dict]
+    :param visible_to: Teams and users that the alert will become visible
+                      to without sending any notification.
+    :type visible_to: list[dict]
+    :param actions: Custom actions that will be available for the alert.
+    :type actions: list[str]
+    :param tags: Tags of the alert.
+    :type tags: list[str]
+    :param details: Map of key-value pairs to use as custom properties of the alert.
+    :type details: dict
+    :param entity: Entity field of the alert that is
+                    generally used to specify which domain alert is related to.
+    :type entity: str
+    :param source: Source field of the alert. Default value is
+                    IP address of the incoming request.
+    :type source: str
+    :param priority: Priority level of the alert. Default value is P3.
+    :type priority: str
+    :param user: Display name of the request owner.
+    :type user: str
+    :param note: Additional note that will be added while creating the alert.
+    :type note: str
+    :param proxy: Proxy to use to make the Opsgenie Alert API call
+    :type proxy: str
+    """
+    def __init__(self,
+                 http_conn_id=None,
+                 api_key=None,
+                 message="",
+                 alias=None,
+                 description=None,
+                 responders=None,
+                 visible_to=None,
+                 actions=None,
+                 tags=None,
+                 details=None,
+                 entity=None,
+                 source=None,
+                 priority=None,
+                 user=None,
+                 note=None,
+                 proxy=None,
+                 *args,
+                 **kwargs
+                 ):
+        super(OpsgenieAlertHook, self).__init__(*args, **kwargs)
+        self.http_conn_id = http_conn_id
+        self.api_key = self._get_api_key(api_key, http_conn_id)
+        self.message = message
+        self.alias = alias
+        self.description = description
+        self.responders = responders
+        self.visible_to = visible_to
+        self.actions = actions
+        self.tags = tags
+        self.details = details
+        self.entity = entity
+        self.source = source
+        self.priority = priority
+        self.user = user
+        self.note = note
+        self.proxy = proxy
+
+    def _get_api_key(self, api_key, http_conn_id):
+        """
+        Given either a manually set key or a conn_id, return the api_key to use
+        :param api_key: The manually provided key
+        :type api_key: str
+        :param http_conn_id: The conn_id provided
+        :type http_conn_id: str
+        :return: api_key (str) to use
+        """
+        if api_key:
```

Comment:
Maybe we should discuss whether pass `key` directly to hook/operater or not, for safe.
like https://github.com/apache/airflow/pull/4895#issuecomment-471385356